crates/sc-proof-of-time/src/verifier.rs

```diff
@@ -44,7 +55,7 @@ impl PotVerifier {
         slot_iterations: NonZeroU32,
         checkpoints: PotCheckpoints,
     ) {
-        self.cache.lock().push(
+        self.cache.lock().0.insert(
```

Comment:
It is better to implement `Deref` and `DerefMut` on such wrappers so you don't need to use `.0`.